### PR TITLE
Conform the style rules of `GeneralUserSettingsTab.tsx` to the style guide

### PR DESCRIPTION
**Pull request by luixxiul**
This PR intends to conform the style rules of general user settings tab to our updated style guide.

type: task

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/components/views/settings/ChangePassword.tsx [hunk 1]
```diff
@@ -399,7 +399,7 @@ export default class ChangePassword extends React.Component<IProps, IState> {
                             <PassphraseField
                                 fieldRef={(field) => (this[FIELD_NEW_PASSWORD] = field)}
                                 type="password"
-                                label={_td("New Password")}
+                                label={_td("New password")}
```
Comment (richvdh) [inline src/components/views/settings/ChangePassword.tsx:402]: I think we should avoid making this change unless we change other references to `New Password` in the application (which should be in a separate PR)

**Review by richvdh — COMMENTED** [review]
@luixxiul: I'm a bit concerned that this is making some quite significant changes to the CSS, and it is hard to review them for correctness. Is there any way we can land the test without changing the CSS so much?

File: src/components/views/settings/ChangePassword.tsx (hunk 1 above)
Comment (luixxiul) [reply src/components/views/settings/ChangePassword.tsx:402]: I would agree. I am going to create an issue to track it.

**Comment by luixxiul** [comment]
> @luixxiul: I'm a bit concerned that this is making some quite significant changes to the CSS, and it is hard to review them for correctness. Is there any way we can land the test without changing the CSS so much?

It is possible, and the test indeed should be landed at first, with a Percy snapshot test which I have forgotten to add. I am going to create another PR which cherry-pick the commit of the test, adjusting class names. Changing CSS rules will be properly tested then. I'm wondering if this would sound fine to you.

**Comment by richvdh** [comment]
It sounds perfect to me!

**Comment by luixxiul** [comment]
The PR for the test was created here: https://github.com/matrix-org/matrix-react-sdk/pull/10658

File: src/components/views/settings/ChangePassword.tsx (hunk 1 above)
Comment (luixxiul) [reply src/components/views/settings/ChangePassword.tsx:402]: The issue was created here: https://github.com/vector-im/element-web/issues/25148. It looks like that capitalization on that area has not been very strict.

**Comment by luixxiul** [comment]
@richvdh  The snapshot ([`User settings tab - General`](https://percy.io/dfde73bd/matrix-react-sdk/builds/26815162/changed/1492765273?browser=firefox&browser_ids=36%2C38%2C39&subcategories=approved&viewLayout=overlay&viewMode=new&width=1024&widths=320%2C640%2C1024%2C1920)) was taken as expected, and now it should be good to restart working on this :+1: I'll ping you after I finish checking the style rules.

File: src/components/views/settings/ChangePassword.tsx (hunk 1 above)
Comment (luixxiul) [reply src/components/views/settings/ChangePassword.tsx]: It was reverted.

File: src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx [hunk 2]
```diff
@@ -454,17 +454,17 @@ export default class GeneralUserSettingsTab extends React.Component<IProps, ISta
...
 
                 <span className="mx_SettingsTab_subheading">{_t("Phone numbers")}</span>
                 {msisdns}
-            </div>
+            </>
         ) : null;
 
         return (
-            <div className="mx_SettingsTab_section">
+            <div className="mx_SettingsTab_section mx_GeneralUserSettingsTab_section--discovery">
```
Comment (luixxiul) [inline src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx:467]: This should not change how `.mx_SettingsTab_subheading` and `.mx_Spinner` are displayed, as both of them are elements inside `threepidSection`, which this element consists.

**Comment by luixxiul** [comment]
It looks like something is wrong with the test. I'm checking.

File: res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss [hunk 3]
```diff
@@ -14,39 +14,47 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-.mx_GeneralUserSettingsTab_changePassword .mx_Field {
-    margin-inline-end: var(--SettingsTab_fullWidthField-margin-inline-end);
+:root {
+    --GeneralUserSettingsTab_subheading-margin-block-start: 24px;
```
Comment (richvdh) [inline res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss:18]: It would be good to add a comment describing what this variable actually means. 

```suggestion
    // The top margin of subheadings within the `GeneralUserSettingsTab`
    --GeneralUserSettingsTab_subheading-margin-block-start: 24px;
```

maybe?

**Review by richvdh — COMMENTED** [review]
looks sensible to me otherwise

File: res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss (hunk 3 above)
Comment (luixxiul) [reply res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss:18]: Thanks, I would agree :+1:

**Comment by luixxiul** [comment]
@richvdh Thank you for the review. It should be fine now.

**Comment by luixxiul** [comment]
https://github.com/matrix-org/matrix-react-sdk/pull/10679 should be merged at first.

**Review by richvdh — APPROVED** [review]

File: res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss [hunk 4]
```diff
@@ -14,54 +14,64 @@ See the License for the specific language governing permissions and
...
-}
+    .mx_GeneralUserSettingsTab_section--account_changePassword {
+        .mx_Field {
+            margin-inline-end: var(--SettingsTab_fullWidthField-margin-inline-end);
 
-.mx_GeneralUserSettingsTab_discovery_existing {
-    display: flex;
-    align-items: center;
-    margin-bottom: 5px;
-}
```
Comment (richvdh) [inline res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss:49]: Following @gsouquet's comments on #10679, these were moved to the top level. It seems wrong that you should nest them now?

Comment (luixxiul) [reply res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss:49]: [Above](https://github.com/matrix-org/matrix-react-sdk/pull/10595/files#diff-8f8bb298ce398e8724d0d757d7ae359c3ee68d3c53f5c82053bc3b05a38eefc6R24) `.mx_GeneralUserSettingsTab_section--discovery_existing` is nested, and setting this `mx_GeneralUserSettingsTab_section--discovery_existing` to a different level can lead to a regression due to the difference of specificity.

This means that `mx_GeneralUserSettingsTab` on [this line above](https://github.com/matrix-org/matrix-react-sdk/pull/10595/files#diff-8f8bb298ce398e8724d0d757d7ae359c3ee68d3c53f5c82053bc3b05a38eefc6R22) should be removed,  in order to remove the difference of specificity altogether.

I am going to push a commit to address this.

Comment (luixxiul) [reply res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss]: Addressed with https://github.com/matrix-org/matrix-react-sdk/pull/10595/commits/4cb0e593ea5951fa15e3dfe9f4a35f32cbc0625f.

Comment (luixxiul) [reply res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss]: And I remember that comment was directed to nesting of `mx_GeneralUserSettingsTab_section--discovery_existing_address` and `mx_GeneralUserSettingsTab_section--discovery_existing_promptText` into `mx_GeneralUserSettingsTab_section--discovery_existing`, so the nesting by this PR was not related to that comment anyway, because this PR was going to nest those selectors' declarations to `mx_GeneralUserSettingsTab`.

**Comment by richvdh** [comment]
This has changed quite a bit since the earlier reviews, so I think it needs a full re-review. I don't think I'll have time today.

**Review by richvdh — CHANGES_REQUESTED** [review]
@luixxiul I'm afraid I'm still struggling to follow this. It's probably fine, but I am conscious of the risk of introducing bugs if anything is incorrect. I appreciate your work in adding a test, but it cannot detect every problem so we still need to review changes like this carefully.

One thing that makes it hard to review is that it seems to be conflating two different types of change into the same PR:
 * Renaming classes like `mx_GeneralUserSettingsTab_accountSection` to `mx_GeneralUserSettingsTab_section--account` to conform to the style guide
 * Restructuring the CSS to collect all the settings for a given class (such as `mx_SetIdServer`) into one place.

Would it be possible to split up this PR to separate the structural changes from the renames?

**Comment by luixxiul** [comment]
I understood. I'll recreate the PR from removing commits for another PR, so please forget the current status of the PR.

**Comment by luixxiul** [comment]
I updated the PR to exclude anything not related to conforming the file to our naming policy.

**Review by richvdh — APPROVED** [review]
Thanks.

I have to say that I'm not convinced this sort of renaming exercise provides much value, compared to the time taken to write and review the changes.

Nevertheless, this looks sensible enough.

**Comment by luixxiul** [comment]
Thanks for the review!

> the time taken to write and review the changes

You would not have to be worried about my side as I am just doing that on my free will for free, following our style guide, to make the style codebase understandable to anyone, as easily as possible.
